rudder-lang/src/translate.rs

```diff
@@ -393,7 +393,8 @@ fn translate_condition(_config: &toml::Value, cond: &str) -> Result<String> {
         // matches a word that can be negative and that be followed by .| + word... possibly wrapped into parenthesis
         static ref OS_RE: Regex = Regex::new(
             // OS part: debian_9_0 \ And optional condition: (!(ubuntu|otheros).something)
-            r"^\(?(?P<os>([a-zA-Z\d]+)(_([a-zA-Z\d]+))*(\|([a-zA-Z\d]+)(_([a-zA-Z\d]+))*)*)\)?(.(?P<cdt>\(\(*!*\(*\w+\)*([.|]\(*!*\(*\w+\)*)*\)))?$"
+            // r"^\(?(?P<os>([a-zA-Z\d]+)(_([a-zA-Z\d]+))*(\|([a-zA-Z\d]+)(_([a-zA-Z\d]+))*)*)\)?(.(?P<cdt>\(\(*!*\(*\w+\)*([.|]\(*!*\(*\w+\)*)*\)))?$"
+            r"^(\(*\w+\(*\)*(\||\.)\(*\w+\)*)*$"
```

Comment:
|| should not appear in a regex, either you mean 
- a litteral || : with \|\|
- an OR operation : with |

Reply:
\||\. => I meant an Or operation : must found either a "|" or "."
What should I do then ?

Reply:
I could rewrite it like this to clarify: `(\\.|\\|)`